docs/reference/settings/security-settings.asciidoc

```diff
@@ -1947,6 +1947,311 @@ include::{es-repo-dir}/settings/common-defs.asciidoc[tag=ssl-supported-protocols
+OpenID Connect Relying Party application during registration
+at the OpenID Connect Provider.
+// end::jwt-allowed-audiences-tag[]
+
+// tag::jwt-claims-principal-tag[]
+`claims.principal`::
+(<<static-cluster-setting,Static>>)
+The name of the JWT claim that contains the user's principal (username).
+// end::jwt-claims-principal-tag[]
+
+// tag::jwt-claims-groups-tag[]
+`claims.groups` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+The name of the JWT claim that contains the user's groups.
+// end::jwt-claims-groups-tag[]
+
+// tag::jwt-claim-pattern-principal-tag[]
+`claim_patterns.principal` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+A Java regular expression that is matched against the JWT claim specified
+by `claims.principal` before it is applied to the user's _principal_ property.
+The attribute value must match the pattern and the value of the first
+_capturing group_ is used as the principal. For example, `^([^@]+)@example\\.com$`
+matches email addresses from the "example.com" domain and uses the local-part as
+the principal.
+// end::jwt-claim-pattern-principal-tag[]
+
+// tag::jwt-claim-pattern-groups-tag[]
+`claim_patterns.groups` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+As per `claim_patterns.principal`, but for the _group_ property.
+// end::jwt-claim-pattern-groups-tag[]
+
+// tag::jwt-populate-user-metadata-tag[]
+`populate_user_metadata` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Specifies whether to populate the {es} user's metadata with the values that are
+provided by the JWT claims. Defaults to `true`.
+// end::jwt-populate-user-metadata-tag[]
+
+// tag::jwt-client-authentication-type-tag[]
+`client_authentication.type` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Specifies whether to use `sharedsecret`, `ssl`, or `none` authentication
+to incoming client requests. Defaults to `sharedsecret`.
+// end::jwt-client-authentication-type-tag[]
+
+`authorization_realms`::
+(<<static-cluster-setting,Static>>)
+The names of the realms that should be consulted for delegated authorization.
+If this setting is used, then the JWT realm does not perform role
+mapping and instead loads the user from the listed realms.
+See <<authorization_realms>>.
+
+`cache.ttl`::
+(<<static-cluster-setting,Static>>)
+Specifies the time-to-live for cached user entries. A user and a hash of its
+credentials are cached for this period of time. Use the
+standard {es} <<time-units,time units>>).
+Defaults to `20m`.
+
+`cache.max_users`::
+(<<static-cluster-setting,Static>>)
+Specifies the maximum number of user entries that the cache can contain.
+Defaults to `100000`.
+
+// tag::jwt-http-connect-timeout-tag[]
+`http.connect_timeout` {ess-icon}::
+(<<static-cluster-setting,Static>>)
+Controls the behavior of the http client used for back-channel communication to
+the JWT Issuer endpoints. Specifies the timeout until a connection
+ is established. A value of zero means the timeout is not used. Defaults to `5s`.
```

Comment:
```suggestion
Controls the behavior of the http client used for fetching the JSON Web Key Set
from remote a URL. A value of zero means the timeout is not used. Defaults to `5s`.
```

Reply:
In OpenID Connect our realm talks to the OP in various endpoints so that wording made sense there. Here we will only reach out for the jwkset url so we can be specific